firmware/src/spi_rgb.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * spi_rgb.c
+ * Copyright (C) 2015 Simon Newton
+ */
+
+#include "spi_rgb.h"
+
+#include "peripheral/spi/plib_spi.h"
+#include "syslog.h"
+
+// TODO(simon): move these into the config (and set with RDM?)
```

Comment:
Yes to the RDM bit. We should define a few manufacturer PIDs which are common between this and OLA. I also think we should split the protocol from the personality. Pixel count and protocol seems like the only stuff we'll need to set via RDM, slots per pixel and latch bytes can be inferred from the protocol.